### PR TITLE
Reviewing a design for a11y post

### PR DESCRIPTION
**Pull request by SteveBarnett**
This Pull Request closes issue #1327.

- Add Steve Barnett (that's me!) to `authors.json`
- Add Reviewing a design for accessiblity post to `posts`

I would really apprecaite feedback on how to make this better! I was considering expanding each bullet point more, but I decided to stay on the side of short(er) and sweet(ish). I did a version with each item as an `h3` but it just seemed... weird.

### CODE REVIEW

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+date: 2021-08-14
+tags:
+  - quicktip
+---
+
+I'm a big believer in [shifting left](https://www.deque.com/shift-left/): considering accessibility as early in the process as we can. One way we can do this is by reviewing designs for accessibility. This helps us catch anything we've missed before we start developing it.
+
+## What to do: annotate
+
+By adding annotations about the semantics and accessibility of elements on a page we can remove barriers to access. The best way to do annotations might be sticky notes on a printout or comments on a Figma file, or something else. The important thing is that we want to be clear and explicit.
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:15]: >comments on a Figma file

Could we link to documentation on how to do this in Figma? I am thinking that some readers may be unfamiliar with the tool.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Here are a handful of things that I've found give good value / effort balance when reviewing a design for accessibility. 
+
+## Things to check
+
+- **Page title**
+    - Is the page title unique?
+    - Does it describe the page content?
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:23]:
```suggestion
    - Does it concisely describe the page content?
```

I am suggesting this if only because my bank took this too literally and dumped the entire intro paragraph into their page title 😓

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- **Headings**
+    - Are headings used liberally?
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:25]: Could we rephrase this to be a little more descriptive? Something along the lines of "headings divide content up in a logical way"? I worry some people might over-correct and provide too many headings.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - Are the levels shown, without skipping any?
+- **Text alternatives**
+    - Do all images (including icons) have a text alternative?
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:28]: >a text alternative?

Could we link to a post somewhere that discusses what an `alt` attribute is and how it works? This is another case where the reader may be unfamiliar.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - Does it describe the page content?
+- **Headings**
+    - Are headings used liberally?
+    - Are the levels shown, without skipping any?
+- **Text alternatives**
+    - Do all images (including icons) have a text alternative? 
+    - If the image is decorative, is it marked as such? Are we sure it's only decorative?
+- **Link and button text**
+    - Does each link describe where we'll go when we follow it? Does each button describe what will happen when we use it?
+    - Where links or buttons have the same text repeated on a page, do they have a longer accessible name that is unique? For example: "View details of Quarter 1 review" instead of "View details".
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:32]: 💯

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+- **Standard controls**
+    - Are standard controls used? They have accessibility baked in. Example: a pair of radio buttons or a checkbox instead of a custom toggle.
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:34]: I think we could turn this into two bullet points, one for identifying if a standard control is applicable, then another for making sure it is utilized. There are situations where a standard control being used for a nonstandard component may create more confusion than it solves (GitHub's short-lived checkbox modal hack comes to mind).

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+- **Names for form elements**
+    - Do all form controls (one input plus one label) have a name?
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:36]: Could we link to a post about how to create the input/label association?

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+- **Link and button text**
+    - Does each link describe where we'll go when we follow it? Does each button describe what will happen when we use it?
+    - Where links or buttons have the same text repeated on a page, do they have a longer accessible name that is unique? For example: "View details of Quarter 1 review" instead of "View details".
+- **Standard controls**
+    - Are standard controls used? They have accessibility baked in. Example: a pair of radio buttons or a checkbox instead of a custom toggle.
+- **Names for form elements**
+    - Do all form controls (one input plus one label) have a name? 
+    - If the name isn't visible, is it shown in an annotation?
+- **Names for groups of form elements**
+    - Do groups of related forms controls (such as multiple choice questions) have a name? The name should provide context for the individual questions in the group.
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:39]: Similarly, can we link to some documentation about the `name` attribute? I can count on one hand the number of developers I've worked with who were aware of needing to go grouping this way.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    - If the name isn't visible, is it shown in an annotation?
+- **Error messages**
+    - Are error messages clear and helpful?
+    - Have we said what went wrong and how to fix it?
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:43]: Maybe also not relying on color alone to indicate error state?

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+## Summary
+
+Reviewing designs for accessibility is a great way to help designers think more accessibly and to help developers implement more accessibly. We can use annotations to be explicit about the semantic and accessibility features we want on each page. We can use short checklists (like the one in this article) to help us make sure we haven't missed any big things.
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accesibility.md:47]: Love it. Maybe we could also work in a link to [our Checklist](https://www.a11yproject.com/checklist/) in our "if you'd like to learn about more you can do" kind of way?

**Review by ericwbailey — CHANGES_REQUESTED** [review]
I think this will be a great addition, thank you for writing it! A couple questions and a few requests and I think we'll be good to publish 🎉

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 1 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Great idea! Added in 236334c.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 3 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Excellent point. I've updated this in 3167ae2.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 4 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: The WebAIM article on `alt` text is my favourite, so I've linked to that and updated the words a bit in c47ad7d.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 10 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Yes! 🥳  Done in 396576d.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 7 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Yes! Added in 98c5964 :)

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 8 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Ooooof! Same. Although by name here I meant accessible name 🙃 . In 0324a06 I've added a link to a WAI article about `fieldset` and `legend` and using `role=group` and `aria-labelledby` as a fallback option.

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 9 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Done in 50add63.

I also added a line about sensory characteristics in 65c760c . Happy to pluck that out if you think it doesn't fit!

File: src/posts/2021-08-14-reviewing-a-design-for-accesibility.md (hunk 6 above)
Comment (SteveBarnett) [reply src/posts/2021-08-14-reviewing-a-design-for-accesibility.md]: Done in 271b29b. I'm not quite happy with what I've got though. Any advice for a better phrasing? 😬

**Comment by SteveBarnett** [comment]
@ericwbailey Okay! Thank you for the very excellent feedback 🙇‍♂️. I've made some changes and left comments on your comments. Could you have another look, please?

I also fixed the typo I made in the filename 🤦‍♂️ .

File: src/posts/2021-08-14-reviewing-a-design-for-accessibility.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    - Are headings used to divide content up in a logical way?
+    - Are the levels shown, without skipping any?
+- **Text alternatives**
+    - Do all images (including icons) have a text alternative (using the `alt` attribute)? (See [WebAIM's guide to alternative text](https://webaim.org/techniques/alttext/) for more about what this is, and what makes good alternative text)
+    - If the image is decorative, is it marked as such (using `alt=""`)? Are we sure it's only decorative?
+- **Link and button text**
+    - Does each link describe where we'll go when we follow it? Does each button describe what will happen when we use it?
+    - Where links or buttons have the same text repeated on a page, do they have a longer accessible name that is unique? For example: "View details of Quarter 1 review" instead of "View details".
+- **Standard controls**
+    - Do form controls match standard ones (like `selects`, `radio`, and `checkbox`) where possible? If not, is it possible to use a standard control instead? (They have accessibility baked in)
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accessibility.md:34]:
```suggestion
    - Do form controls match standard ones (like `select`s, `radio`, and `checkbox`) where possible? If not, is it possible to use a standard control instead?
```

Two things here:

1. Removing "s" from the `select` element's name and moving it outside.
2. Removing the "accessibility baked in" line because if you get suuuuper technical that's not always true. I think the fact that it's posted here implies it'll be net-better for assistive technology support, though.

File: src/posts/2021-08-14-reviewing-a-design-for-accessibility.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+## Things to check
+
+- **Page title**
+    - Is the page title unique?
+    - Does it concisely describe the page content?
+- **Headings**
+    - Are headings used to divide content up in a logical way?
+    - Are the levels shown, without skipping any?
+- **Text alternatives**
+    - Do all images (including icons) have a text alternative (using the `alt` attribute)? (See [WebAIM's guide to alternative text](https://webaim.org/techniques/alttext/) for more about what this is, and what makes good alternative text)
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accessibility.md:28]:
```suggestion
    - Do all images (including icons) have a text alternative (using the `alt` attribute)? You can see [WebAIM's guide to alternative text](https://webaim.org/techniques/alttext/) for more about what this is, and what makes good alternative text.
```

I think we can promote this from a parenthetical into a full-fledged statement!

File: src/posts/2021-08-14-reviewing-a-design-for-accessibility.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+    - If the image is decorative, is it marked as such (using `alt=""`)? Are we sure it's only decorative?
+- **Link and button text**
+    - Does each link describe where we'll go when we follow it? Does each button describe what will happen when we use it?
+    - Where links or buttons have the same text repeated on a page, do they have a longer accessible name that is unique? For example: "View details of Quarter 1 review" instead of "View details".
+- **Standard controls**
+    - Do form controls match standard ones (like `selects`, `radio`, and `checkbox`) where possible? If not, is it possible to use a standard control instead? (They have accessibility baked in)
+    - Are standard controls being used to match the design intent?
+- **Names for form elements**
+    - Do all form controls (one input plus one label) have a name? (See [MDN's article on the `<label>` HTML element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for an example of how the `label` and `input` elements work together with the `id` and `for` attributes to provide accessible names)
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accessibility.md:37]:
```suggestion
    - Do all form controls (one input plus one label) have a name? You can reference [MDN's article on the `<label>` HTML element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) for an example of how the `label` and `input` elements work together with the `id` and `for` attributes to provide accessible names.
```

Another good sentence here. Also suggesting changing "see" to "reference" , as not everyone can see.

File: src/posts/2021-08-14-reviewing-a-design-for-accessibility.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+    - If the name isn't visible, is it shown in an annotation?
+- **Names for groups of form elements**
+    - Do groups of related forms controls (such as multiple choice questions) have a name? The name should provide context for the individual questions in the group. (See [the W3C's Web Accessibility Tutorial on Grouping Controls](https://www.w3.org/WAI/tutorials/forms/grouping/) for two approaches to this)
```
Comment (ericwbailey) [inline src/posts/2021-08-14-reviewing-a-design-for-accessibility.md:40]:
```suggestion
    - Do groups of related forms controls (such as multiple choice questions) have a name? The name should provide context for the individual questions in the group. Reference [the W3C's Web Accessibility Tutorial on Grouping Controls](https://www.w3.org/WAI/tutorials/forms/grouping/) for two approaches on how to do this.
```

Another sentence!

**Review by ericwbailey — CHANGES_REQUESTED** [review]
Great stuff—a few more things and we're good to publish! We'll also announce the post on Twitter and our newsletter 🎉

Once you make these changes, could you send us a message via [our contact form](https://www.a11yproject.com/contact/) for payment?

**Comment by SteveBarnett** [comment]
Woo! 🥳 
Changes made. Thank you for those updates, they're great! 😄

**Review by ericwbailey — APPROVED** [review]
Great! Thank you for working with the feedback process, this should be live tomorrow 🎉
